org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DocumentLifeCycleHandler.java

```diff
@@ -325,12 +341,33 @@ public void handleChanged(DidChangeTextDocumentParams params) {
 				} else {
 					edit = new ReplaceEdit(startOffset, length, text);
 				}
-				IDocument document = JsonRpcHelpers.toDocument(unit.getBuffer());
-				edit.apply(document, TextEdit.NONE);
+
+				// Avoid any computation iff the `SemanticHighlightingService#isEnabled` is `false`.
```

Comment:
Just for completeness with the [source](https://en.wikipedia.org/wiki/If_and_only_if):
"[...]_In logic and related fields such as mathematics and philosophy, if and only if (shortened **iff**) is a biconditional logical connective between statements._[...]"

Since it was not obvious whether it is a typo or not, I apply the recommended change. Thanks for the feedback.